### PR TITLE
Implement USB for samd51, particularly the itsybitsy_m4

### PR DESCRIPTION
**Pull request by twitchyliquid64**
Not yet 100% ready to go, but wanted to see what travis CI would do.

This is mostly based on the samd21 implementation, but updated for the newer usb_device implementation, different NVIC, different clocking etc.

Comments welcome.

### CODE REVIEW

**Comment by sajattack** [comment]
Looks great! Could you update samd21 as well?

**Comment by twitchyliquid64** [comment]
I would rather do that in a separate PR. I don't yet have a SAMD21 device.

**Comment by sajattack** [comment]
Fair enough.

**Comment by sajattack** [comment]
Let me know when you think it's ready for merge.

File: hal/src/lib.rs [hunk 1]
```diff
@@ -48,15 +47,17 @@ macro_rules! dbgprint {
...
 pub use self::samd51::*;
 
 #[cfg(not(feature="samd51"))]
 pub mod samd21;
 #[cfg(not(feature="samd51"))]
 pub use self::samd21::*;
 
-#[cfg(feature = "usb")]
+#[cfg(all(feature = "usb", feature="samd21"))]
 pub use self::samd21::usb;
```
[the remark this answers is not in the document]
Comment (twitchyliquid64) [reply hal/src/lib.rs]: Hmmm, I thought this `cfg` clause means to only re-export the usb module if both the samd21 and usb features are selected? This mirrors the previous behavior.

None of the board support crates enable the usb feature, so it shouldnt become available. They have lines commented out to enable it however.

[the remark this answers is not in the document]
Comment (sajattack) [reply hal/src/lib.rs]: Oh right.

**Comment by twitchyliquid64** [comment]
I think this is ready for review. A few comments:

 * I'm not sure of the best way to setup a UART for use with the `dbgprint!` macro. The `uart_debug` thing / feature is what I came up with, but I don't like it and would defs like to change it to something more elegant if you have any ideas.

 * I haven't yet incremented any versions in the cargo files. Reckon this is a minor or major semver change?

**Comment by sajattack** [comment]
If it doesn't break anything it would be a minor revision I think.

**Comment by twitchyliquid64** [comment]
Okies, I've modified the feature name to match the convention established by `use_rtt` (`uart_debug` --> `use_uart_debug`), but otherwise I've leave it as-is.

**Comment by twitchyliquid64** [comment]
Could you push out updated crate versions for samd51-hal and itsybitsy_m4? :)

**Comment by sajattack** [comment]
Done.

**Pull request merged**